src/twin/mod.rs

```diff
@@ -270,52 +272,65 @@ impl Twin {
     }
 
     async fn handle_connection_status(&mut self, auth_status: AuthenticationStatus) -> Result<()> {
-        let mut online = false;
-
         match auth_status {
             AuthenticationStatus::Authenticated => {
-                if !self.authenticated_once {
+                if !self.authenticated {
                     info!("Succeeded to connect to iothub");
 
-                    /*
-                     * the update validation test "wait_for_system_running" enforces that
-                     * omnect-device-service already notified its own success
-                     */
-                    self.update_validation.set_authenticated().await?;
+                    if !self.authenticated_once {
+                        /*
+                         * the update validation test "wait_for_system_running" enforces that
+                         * omnect-device-service already notified its own success
+                         */
+                        self.update_validation.set_authenticated().await?;
 
-                    self.connect_twin().await?;
+                        self.authenticated_once = true;
+                    };
```

Comment:
why do we use a boolean here and have to distinguish between `authenticated` and `authenticated_once`?
why not `call_once`?

Reply:
afaik you cannot use async within call_once closure